File: src/main/java/com/akash/projects/airBnbApp/service/BookingService.java
package com.akash.projects.airBnbApp.service;

import com.akash.projects.airBnbApp.dto.BookingDto;

import java.util.List;

public interface BookingService {
    BookingDto createBooking(Long hotelId, Long roomId, BookingDto bookingDto);
    BookingDto getBookingById(Long bookingId);
    List<BookingDto> getAllBookingsForUser();
    void cancelBooking(Long bookingId);
}

File: airbnb-frontend/src/services/bookingService.js
import api from './api'

export const bookingService = {
  createBooking: (hotelId, roomId, bookingData) => 
    api.post(`/bookings/hotels/${hotelId}/rooms/${roomId}`, bookingData),
  getBookingById: (id) => api.get(`/bookings/${id}`),
  getMyBookings: () => api.get('/bookings/my-bookings'),
  cancelBooking: (id) => api.patch(`/bookings/${id}/cancel`),
}
